transit_odp/pipelines/pipelines/dataset_etl/utils/transform.py

```diff
@@ -68,6 +92,25 @@ def agg_service_pattern_sequences(df: pd.DataFrame):
     )
 
 
+def agg_flexible_service_pattern_sequences(df: pd.DataFrame):
+    points = df["geometry"].values
+    geometry_points = [
+        [point.x, point.y]
+        for point_list in points
```

Comment:
Could we check if `explode()` dataframe function can be used here to avoid nested loop